pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -157,19 +118,19 @@ var _ directbase.Adapter = &Adapter{}
 func (a *Adapter) Find(ctx context.Context) (bool, error) {
 	log := klog.FromContext(ctx).WithName(ctrlName)
 
-	log.V(2).Info("getting BigQueryConnectionConnection", "name", a.id.AsExternalRef())
+	log.V(2).Info("getting BigQueryConnectionConnection", "name", a.desired.Name)
 
-	if a.id.serviceGeneratedID == "" {
+	if a.id.External == "" {
 		// Cannot retrieve the Connection without ServiceGeneratedID, expecting to create a new Connection.
 		return false, nil
 	}
-	req := &bigqueryconnectionpb.GetConnectionRequest{Name: a.id.FullyQualifiedName()}
+	req := &bigqueryconnectionpb.GetConnectionRequest{Name: a.id.External}
 	connectionpb, err := a.gcpClient.GetConnection(ctx, req)
 	if err != nil {
 		if direct.IsNotFound(err) {
 			return false, nil
 		}
-		return false, fmt.Errorf("getting BigQueryConnectionConnection %q: %w", a.id.FullyQualifiedName(), err)
+		return false, fmt.Errorf("getting BigQueryConnectionConnection %q: %w", a.desired.Name, err)
```

Comment:
I don't think we normally populate a.desired.Name, at least not in the default mappers..